packages/playwright-core/src/server/electron/electron.ts

```diff
@@ -195,7 +195,11 @@ export class Electron extends SdkObject {
         tracesDir: artifactsDir,
       };
       validateBrowserContextOptions(contextOptions, browserOptions);
-      const browser = await CRBrowser.connect(chromeTransport, browserOptions);
+      const browser = await Promise.race([
+        waitForLine(progress, launchedProcess, /^Waiting for the debugger to disconnect...$/),
+        CRBrowser.connect(chromeTransport, browserOptions),
+      ]);
+      if (!browser || !(browser instanceof CRBrowser)) throw new Error('canceled');
```

Comment:
This looks wrong: as far as the client is concerned, Electron was launched, it's main js ran w/o errors. It might have done things and we could even try testing it's main context.

Reply:
What I tried to fix was that the `CRBrowser.connect(chromeTransport, browserOptions)` promise never resolved

Inside `CRBrowser.connect` is this code: `await session.send('Browser.getVersion');` which never resolves

![code](https://user-images.githubusercontent.com/23744935/159759854-12b987d3-a75d-4722-804b-9d9801b870ea.png)


What would be another solution to this problem?

Reply:
@SimonSiefke My only concern is that a seemingly successful launch should not lead to exception. I.e. we should handle early exit of Electron gracefully. Giving a more detailed hint would require me to look at the code and think for a while - it ain't easy task.